ansible_base/lib/testing/fixtures.py

```diff
@@ -42,6 +43,33 @@ def _env(key, value):
     return _env
 
 
+@pytest.fixture
+def settings_override_mutable(settings):
+    """
+    pytest-django's settings  doesn't handle mutable settings types well
+    (https://github.com/pytest-dev/pytest-django/issues/601). This fixture is an
+    attempt to solve that problem.
+
+    Use as follows:
+
+    def test_foo(settings, settings_override_mutable):
+        with settings_override_mutable("SOME_SETTING_VAR"):
+            settings.SOME_SETTING_VAR['and']['nested']['a']['bit'] = "new_value"
+            # or even
+            delattr(settings, 'SOME_SETTING_VAR')  # for some reason normal "del" doesn't work here
+        # But out of the context manager the whole var is restored
+        assert settings.SOME_SETTING_VAR['and']['nested']['a']['bit'] == "old_value"
+    """
+
+    @contextmanager
+    def f(setting_key):
+        original = copy.deepcopy(getattr(settings, setting_key))
+        yield
```

Comment:
the docs say you should have a try-except

https://docs.python.org/3/library/contextlib.html

Reply:
good catch, I fixed a few other fixtures too.